docs/book/src/cronjob-tutorial/pprof-tutorial.md

```diff
@@ -0,0 +1,59 @@
+# Using Pprof for Controller Profiling
+
+Pprof is a Go based library designed for profiling the binary in order to gather CPU, and Memory related data. Pprof is help in finding any sort of bottle necks that are either appearing at Memory, CPU, or both levels.
+
+Pprof itself runs as a tiny HTTP server as part of the the Controller. You can extract the statistics, and visualize them on an interactive website. Furthermore, **you don't need to install this package seperately**. Pprof comes packaged as part of Kubebuilder `controller-runtime` [dependency](https://github.com/kubernetes-sigs/controller-runtime/pull/1943) within the scafolded project.
+
+For further information about the Pprof usage, check out the official [docs](https://github.com/google/pprof).
+
+# How to use Pprof in your Codebase
+
+1. In your `cmd/main.go` file, specify the `PprofBindAddress` URL field as part of Controller Manager's configurations:
+
+    ```golang
+    mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+      ....
+      Scheme:                 scheme,
+      LeaderElection:         enableLeaderElection,
+      PprofBindAddress:       "127.0.0.1:8082", // Specify your own URL here.
```

Comment:
it it not url, it is host interface on which server will listening

Reply:
@grzesuav what's the difference exactly? Network Interface sounds more like NIC (Network Interface Card) which the end user doesn't usually care about.

Reply:
We need add a valid comment/term, see the definition in the lib:

https://github.com/kubernetes-sigs/controller-runtime/blob/38546806f2faf5973e3321a7bd5bb3afdbb5767d/pkg/manager/manager.go#L237-L242

So, I would suggest
```suggestion
      // PprofBindAddress is the TCP address that the controller should bind to
      // for serving pprof. Specify the manager address and the port that should be bind.
      PprofBindAddress:       "127.0.0.1:8082", 
```

Reply:
Sounds reasonable.

Reply:
each network interface has ip address bind to it,from which to are special - like `127.0.0.1` is localhost and `0.0.0.0` means - bind to all local interfaces. I am not a network engineer so my explanation can be a little dump, but think if computer has more than one network card, each ot the network cards is identified by different ip address locally, so you can expose service on one network and not expose on the another.